src/alias/azext_alias/argument.py

```diff
@@ -0,0 +1,165 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=import-error
+
+import re
+import shlex
+
+from knack.util import CLIError
+
+from jinja2 import Template
+from azext_alias._const import (
+    POS_ARG_NAME_REGEX,
+    NUMBER_PLACEHOLDER_REGEX,
+    DUPLICATED_PLACEHOLDER_ERROR,
+    RENDER_TEMPLATE_ERROR,
+    INSUFFICIENT_POS_ARG_ERROR,
+    PLACEHOLDER_EVAL_ERROR,
+    EXPRESSION_REGEX
+)
+
+
+def get_pos_args_names(arg):
+    """
+    Get all the positional arguments' names in order.
+
+    Args:
+        arg: The word which this function performs searching on.
+
+    Returns:
+        A list of positional arguments' names.
+    """
+    pos_args_name = list(map(str.strip, re.findall(POS_ARG_NAME_REGEX, arg)))
```

Comment:
Overall, the map is discouraged in the situations where a list comprehension can accomplish the same task. 
- https://stackoverflow.com/questions/1247486/python-list-comprehension-vs-map